Migrate pkg/kubelet/winstats to contextual logging

Chulong-Li opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?
/kind cleanup
<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->

#### What this PR does / why we need it:
Migrate pkg/kubelet/winstats to contextual logging

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Relates to https://github.com/kubernetes/kubernetes/issues/130069

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
Migrate pkg/kubelet/winstats to contextual logging
```

Chulong-Li committed:
Migrate pkg/kubelet/winstats to contextual logging

Update the order of imports to follow the standard convention

Quick update on import order for cadvisor_windows.go

Update the hack files

Update contextual logging in Test_convertWinApiToCadvisorApi

Chulong-Li commented:
/wg structured-logging
/area logging
/kind cleanup
/cc @kubernetes/wg-structured-logging-reviews

Chulong-Li commented:
/cc @bart0sh

HirazawaUi commented:
/ok-to-test

HirazawaUi commented on pkg/kubelet/winstats/cpu_topology_test.go:
```diff
@@ -20,10 +20,12 @@ limitations under the License.
+
+	cadvisorapi "github.com/google/cadvisor/info/v1"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/klog/v2/ktesting"
```
A super nit: please group the imports by organization. We generally refer to the following convention:

Imports are organized in groups, with blank lines between them. The standard library packages are always in the first group.

ref: https://go.dev/wiki/CodeReviewComments#imports

Although the lint in the k/k repo won’t mind it :)

Chulong-Li replied:
@HirazawaUi Thanks for comments! That's a good point. Should we update it like below?
```go
import (
	"testing"
	"unsafe"

	cadvisorapi "github.com/google/cadvisor/info/v1"
	"github.com/stretchr/testify/assert"

	"k8s.io/klog/v2/ktesting"
)
```

HirazawaUi replied:
Yes, that's quite right.

Chulong-Li replied:
@HirazawaUi Have updated the order of imports for all related files to follow the standard convention. Could you please take another look?

HirazawaUi commented on pkg/kubelet/winstats/network_stats_test.go:
```diff
@@ -25,6 +25,7 @@ import (
 
 	cadvisorapi "github.com/google/cadvisor/info/v1"
 	"github.com/stretchr/testify/assert"
+	"k8s.io/klog/v2/ktesting"
```
Same as above.

Chulong-Li replied:
Updated as well

Chulong-Li commented:
/retest

Chulong-Li commented:
/retest

bart0sh commented:
@Chulong-Li Thank you for the PR! It looks good to me from the first brief look. Please, update lint configurations in hacks/ thanks!

bart0sh commented:
/triage accepted
/priority important-longterm
/assign

Chulong-Li commented:
@bart0sh Thanks for the reminder! Almost missed this part. I've just updated the lint configurations in hack folder. Please take another look. 😊

Chulong-Li commented:
Squashed the commits for simplicity

Chulong-Li commented:
/retest

Chulong-Li commented:
/retest

HirazawaUi commented:
The failure of the `pull-kubernetes-e2e-capz-windows-master` job does not seem to be caused by the current PR. This job has a very low success rate and is not a "**Required**" test job, so you don’t need to repeatedly test it.

bart0sh commented:
/skip pull-kubernetes-e2e-capz-windows-master

bart0sh commented on pkg/kubelet/winstats/cpu_topology.go:
```diff
@@ -233,7 +234,7 @@ func convertWinApiToCadvisorApi(buffer []byte) (int, int, []cadvisorapi.Node, er
 			}
 
 		default:
-			klog.V(4).Infof("Not using Windows CPU relationship type: %d", info.Relationship)
```
It's interesting that this non-structured logging was here. Theoretically it should raise lint errors as all pkg/kubelet [is supposed to use structured logging](https://github.com/kubernetes/kubernetes/blob/master/hack/logcheck.conf#L18).

pohly replied:
We have a gap with linting: it doesn't run for Windows-only code like this one here. Might be worth closing?

cc @BenTheElder @aojea

BenTheElder replied:
We also don't for e.g. GOOS=arm64, right?

We have some very specific linting for "does it typecheck" that targets all platforms, with a custom tool.

I'd be a bit concerned about the cost to golangci-lint for all platforms, but I suppose we're more likely to have build tagged files for linux,windows generally speaking.

pohly replied:
It's definitely gradual and a question of "bang for the buck". My expectation is that we have more Windows-only code than arm64-only code.

pohly replied:
I had a brief look by setting GOOS=windows in the script. There are several valid reports where Windows code is not following conventions that are enforced when building for Linux. I wonder whether SIG Windows is aware.

Anyway, I'm not going to pursue this further. If I did, I would have to fix all those issues...

bart0sh commented:
@Chulong-Li Please, rebase on top of the latest master branch. This can potentially fix CI job failures.

Chulong-Li commented:
> The failure of the `pull-kubernetes-e2e-capz-windows-master` job does not seem to be caused by the current PR. This job has a very low success rate and is not a "**Required**" test job, so you don’t need to repeatedly test it.

@HirazawaUi Thanks for this information! This kind of failure should be transient.

Chulong-Li commented:
@bart0sh Thanks for all these comments! I've just updated and rebased the code. Hope it would fix that check failure issue. Could you please take another look?

Chulong-Li commented:
/retest

bart0sh commented on pkg/kubelet/winstats/cpu_topology_test.go:
```diff
@@ -312,7 +315,8 @@ func Test_convertWinApiToCadvisorApi(t *testing.T) {
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			numOfCores, numOfSockets, nodes, err := convertWinApiToCadvisorApi(tt.buffer)
+			logger, _ := ktesting.NewTestContext(t)
```
In my opinion it should be done in the beginning of the function because it's a test(not a test case) context.

Chulong-Li replied:
@bart0sh That make senses. Thanks for explanation!

bart0sh commented:
/retest

bart0sh commented:
/lgtm

/assign @mrunalp @SergeyKanzhelev 
for approval

HirazawaUi reviewed (commented):
/lgtm

HirazawaUi commented:
/remove-lifecycle stale

Chulong-Li commented:
Thanks @HirazawaUi for reviewing!

Chulong-Li commented:
Hi @SergeyKanzhelev, could you please help review this PR for approval? Thanks in advance!

Chulong-Li commented:
/test pull-kubernetes-e2e-gce

Chulong-Li commented:
Hi @bart0sh, I double checked the pull-kubernetes-unit-windows-master test and had no clue on how dynamicresources issue is related to winstats parts. Do you have any thoughts?
```
=== FAIL: pkg/scheduler/framework/plugins/dynamicresources TestPlugin/timeout/filter/worker (0.00s)
    dynamicresources_test.go:1160: 
        	Error Trace:	C:/kubernetes/pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go:1263
        	            				C:/kubernetes/pkg/scheduler/framework/plugins/dynamicresources/dynamicresources_test.go:1160
        	Error:      	Error "cannot allocate all claims" does not contain "timed out trying to allocate devices"
        	Test:       	TestPlugin/timeout/filter/worker
```

SergeyKanzhelev reviewed (approved):
/lgtm
/approve
/retest-required

bart0sh commented:
/skip

/assign @pohly 
for hack/ changes approval

SergeyKanzhelev commented:
/cc @dims
